crates/driver/src/domain/competition/auction.rs

```diff
@@ -75,39 +80,124 @@ impl Auction {
         &self.orders
     }
 
-    /// Prioritize the orders such that those which are more likely to be
-    /// fulfilled come before less likely orders. Filter out orders which
-    /// the trader doesn't have enough balance to pay for.
-    ///
-    /// Prioritization is skipped during quoting. It's only used during
-    /// competition.
-    pub async fn prioritize(self, eth: &Ethereum, balances: &eth::balances::Cache) -> Self {
-        // Sort orders so that most likely to be fulfilled come first.
-        let mut auction = self;
-        let sort = || {
-            auction.orders.sort_by_cached_key(|order| {
-                // Market orders are preferred over limit orders, as the expectation is that
-                // they should be immediately fulfillable. Liquidity orders come last, as they
-                // are the most niche and rarely used.
-                let class = match order.kind {
-                    competition::order::Kind::Market => 2,
-                    competition::order::Kind::Limit { .. } => 1,
-                    competition::order::Kind::Liquidity => 0,
-                };
-                std::cmp::Reverse((
-                    class,
-                    // If the orders are of the same kind, then sort by likelihood of fulfillment
-                    // based on token prices.
-                    order.likelihood(&auction.tokens),
-                ))
-            });
-            auction
-        };
-        let mut auction = tokio::task::spawn_blocking(sort).await.unwrap();
-
-        let balances = balances.get_or_fetch(eth, auction.orders()).await;
-        let mut balances = (*balances).clone();
+    /// The tokens used in the auction.
+    pub fn tokens(&self) -> &Tokens {
+        &self.tokens
+    }
+
+    /// Returns a collection of liquidity token pairs that are relevant to this
+    /// auction.
+    pub fn liquidity_pairs(&self) -> HashSet<liquidity::TokenPair> {
+        self.orders
+            .iter()
+            .filter_map(|order| match order.kind {
+                order::Kind::Market | order::Kind::Limit { .. } => {
+                    liquidity::TokenPair::new(order.sell.token, order.buy.token).ok()
+                }
+                order::Kind::Liquidity => None,
+            })
+            .collect()
+    }
+
+    pub fn gas_price(&self) -> eth::GasPrice {
+        self.gas_price
+    }
+
+    pub fn deadline(&self) -> Deadline {
+        self.deadline
+    }
+
+    pub fn score_cap(&self) -> Score {
+        self.score_cap
+    }
+}
+
+#[derive(Clone)]
+pub struct PreProcessor(Arc<Mutex<Inner>>);
```

Comment:
nit: Maybe `AuctionProcessor`

Reply:
I thought this is fine since we tend to use namespaces a lot more in the new code and therefore also reference types as such.
In this case that would result in using the type externally as `auction::AuctionProcessor`.